### DISCUSSION TITLE
Clean exit of SnpOnly.efi and Read-Only RamDisk created by SnpOnly.efi

### DISCUSSION THREAD

**Discussion by k-s-srikanth** [ipxe/ipxe#236] [Q&A]
We were booting using IPXE boot on Dell Laptops.  Idea was to download few drivers from the remote and load them. 
Then unload the SnpOnly.efi and the ReadOnly RamDisk created by SnpOnly.efi
Then continue with our Private (FileSystem) driver at UEFI Shell.

a.	Boot from IPXE NBP (network boot program) snponly.efi
b.	Snponly.efi creates Read-Only RAM disk and download all files and boot to UEFI shell from the RAM disk
c.	Run Our Private RamDisk Driver to create a new RAM disk and lets call this Dell RAM disk.
d.     Copy files from snponly.efi Read-Only RAM disk to Dell RAM disk
e.	Unload snponly.efi for releasing TCP/IP stack and reconnect UEFI BIOS TCP/IP stack
f.	Run ifconfig to obtain IP address through DHCP
g.	Run our FS driver. [we have an App that invoke Driver Internally] This creates new FileSystems and these get mounted.
h.     To reflect these mount point and create FS mapping, "map -u" command is executed.
i.	But that results in "map -u got error: status not found" error output.

**Comment by mcb30** [comment]
You have a very complicated test setup.  Please try cutting out steps until you find the action that is responsible for your problem.  For example: do you still see the problem if you omit the "Run our FS driver" step?  If so, then the problem is most likely within your custom filesystem driver.

**Reply by k-s-srikanth** [reply]
We have using the FS driver from an Attached USB, then everything works fine. Only when downloaded from IPXE and Snponly.efi is unloaded, we get into this issue.
Even "unload <handle of snponly.efi> is failing with "Invalid Param" Error.
Meanwhile I will try the experiment suggested by you. Thanks.
